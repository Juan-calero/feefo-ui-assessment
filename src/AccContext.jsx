import { createContext } from "react"

const AccContext = createContext()

export default AccContext
