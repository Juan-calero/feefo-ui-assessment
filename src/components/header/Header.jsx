import React from "react"
import styled from "styled-components"
import Support from "./support/Support"

const StyledHeader = styled.div`
  display: flex;
  & h1 {
    flex: 3;
    font-weight: normal;
    color: #3f3c3d;
  }
  @media (max-width: 599px) {
    flex-direction: column;
    margin: 0rem 1rem 2rem;
    & h1 {
      margin-bottom: 1.7rem;
    }
  }
  @media (min-width: 600px) {
    align-items: center;
    margin-bottom: 2.35rem;
  }
`

function Header() {
  return (
    <StyledHeader>
      <h1>Account Overview</h1>
      <Support />
    </StyledHeader>
  )
}
export default Header
