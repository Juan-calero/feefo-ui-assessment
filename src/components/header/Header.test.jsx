import { render, screen } from "@testing-library/react"
import AccContext from "../../AccContext"
import Header from "./Header"

const accountData = {
  supportContact: {
    name: "John Smith",
    email: "john.smith@feefo.com",
    number: "020 3362 4208",
  },
}

const header = (
  <AccContext.Provider value={accountData}>
    <Header />
  </AccContext.Provider>
)

test("Renders Header", () => {
  expect(render(header)).toMatchSnapshot()
})
test("Displays Header title: `Account Overview`", () => {
  render(header)
  const element = screen.getByText(/Account Overview/i)
  expect(element).toBeInTheDocument()
})
