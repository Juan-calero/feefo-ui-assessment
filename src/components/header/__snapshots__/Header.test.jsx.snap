// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt fvmUtc"
      >
        <h1>
          Account Overview
        </h1>
        <section
          class="sc-dlfnbm iVEacc"
        >
          <h2>
            YOUR FEEFO SUPPORT CONTACT
          </h2>
          <div
            class="card"
          >
            <aside
              class="sc-gsTCUz loiJso"
            >
              JS
            </aside>
            <div
              class="sc-bdfBwQ BzuyY"
            >
              <h3>
                John Smith
              </h3>
              <div
                class="contacts"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope fa-w-16 icon"
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                    fill="currentColor"
                  />
                </svg>
                <a
                  href="mailto:john.smith@feefo.com"
                  rel="noopener noreferrer"
                >
                  john.smith@feefo.com
                </a>
                <a
                  href="tel:02033624208"
                  rel="noopener noreferrer"
                >
                  020 3362 4208
                </a>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt fvmUtc"
    >
      <h1>
        Account Overview
      </h1>
      <section
        class="sc-dlfnbm iVEacc"
      >
        <h2>
          YOUR FEEFO SUPPORT CONTACT
        </h2>
        <div
          class="card"
        >
          <aside
            class="sc-gsTCUz loiJso"
          >
            JS
          </aside>
          <div
            class="sc-bdfBwQ BzuyY"
          >
            <h3>
              John Smith
            </h3>
            <div
              class="contacts"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-envelope fa-w-16 icon"
                data-icon="envelope"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                  fill="currentColor"
                />
              </svg>
              <a
                href="mailto:john.smith@feefo.com"
                rel="noopener noreferrer"
              >
                john.smith@feefo.com
              </a>
              <a
                href="tel:02033624208"
                rel="noopener noreferrer"
              >
                020 3362 4208
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
